lib/blueprinter/formatters/date_time_formatter.rb

```diff
@@ -16,7 +16,16 @@ def format(value, options)
 
     def format_datetime(value, field_format)
       format = field_format || Blueprinter.configuration.datetime_format
-      format.nil? ? value : value.strftime(format)
+
+      if format.nil?
+        value
+      elsif format.is_a?(Proc)
+        format.call(value)
+      elsif format.is_a?(String)
+        value.strftime(format)
+      else
+        raise BlueprinterError, 'Cannot format DateTime object with invalid formatter'
```

Comment:
We ought to raise a more specific error since this is exclusively in the date time formatter domain.

Reply:
Also maybe a message such as `"Cannot format DateTime object with invalid formatter: #{format.class}"`